dev/io.openliberty.security.jakartasec.3.0.internal_fat/fat/src/io/openliberty/security/jakartasec/fat/tests/TokenValidationTests.java

```diff
@@ -0,0 +1,1219 @@
+
+        JWTTokenBuilder builder = initTokenBuilder();
+        // override issuer value with what the OP would return in discovery, but our annotation overrides that value with a different value
+        builder.setClaim(PayloadConstants.PAYLOAD_AUDIENCE, Arrays.asList("badAudience", "client_1"));
+        builder.setClaim(PayloadConstants.PAYLOAD_AUTHORIZED_PARTY, "badAudience");
+        saveTokenEndpointToken(builder);
+
+        runEndToEndTest("TokenTestAppServlet", getAudiencetMismatchExpectations());
+
+    }
+
+    /**
+     * Test that a token returned from the token endpoint without an exp claim will fail validation.
+     */
+    @Test
+    public void TokenValidationTests_tokens_expMissing() throws Exception {
+
+        JWTTokenBuilder builder = initTokenBuilder(PayloadConstants.PAYLOAD_EXPIRATION_TIME_IN_SECS);
+        saveTokenEndpointToken(builder);
+
+        runEndToEndTest("TokenTestAppServlet", getMissingClaimExpectations(PayloadConstants.PAYLOAD_EXPIRATION_TIME_IN_SECS));
+
+    }
+
+    /**
+     * Test that a token returned from the token endpoint with an exp claim value that is in the past will fail validation.
+     */
+    @Test
+    public void TokenValidationTests_tokens_expInThePast() throws Exception {
+
+        JWTTokenBuilder builder = initTokenBuilder(PayloadConstants.PAYLOAD_EXPIRATION_TIME_IN_SECS);
+        builder.setExpirationTime(NumericDate.fromSeconds(1672531140)); // 12/31/2022 11:59:00 PM
+        saveTokenEndpointToken(builder);
+
+        runEndToEndTest("TokenTestAppServlet", getExpInPastExpectations());
+
+    }
+
+    /**
+     * Test that a token returned from the token endpoint with an exp claim value that valid (in the future) and will pass validation
+     */
+    @Test
+    public void TokenValidationTests_tokens_expValid() throws Exception {
+
+        JWTTokenBuilder builder = initTokenBuilder();
+        saveTokenEndpointToken(builder);
+
+        runEndToEndTest("TokenTestAppServlet");
+
+    }
+
+    /**
+     * Test that a token returned from the token endpoint without an iat claim will fail validation.
+     */
+    @Test
+    public void TokenValidationTests_tokens_iatMissing() throws Exception {
+
+        JWTTokenBuilder builder = initTokenBuilder(PayloadConstants.PAYLOAD_ISSUED_AT_TIME_IN_SECS);
+        saveTokenEndpointToken(builder);
+
+        runEndToEndTest("TokenTestAppServlet", getMissingClaimExpectations(PayloadConstants.PAYLOAD_ISSUED_AT_TIME_IN_SECS));
+
+    }
+
+    /**
+     * Test that a token returned from the token endpoint with an iat claim value that is in the future will fail validation.
+     */
+    @Test
+    public void TokenValidationTests_tokens_iatInTheFuture() throws Exception {
+
+        JWTTokenBuilder builder = initTokenBuilder(PayloadConstants.PAYLOAD_ISSUED_AT_TIME_IN_SECS);
+        builder.setIssuedAt(NumericDate.fromSeconds(1735689540)); // 12/31/2024 11:59:00 PM
```

Comment:
It'd be better to set it programmatically to something like 1 day in the future. Then we won't have to deal with a hard build failure that seemingly randomly starts happening around Jan 1, 2025.
```suggestion
        builder.setIssuedAt(NumericDate.fromSeconds(System.currentTimeMillis() + (24 * 60 * 60 * 1000))); // 12/31/2024 11:59:00 PM
```

Reply:
updated - was originally going to set it out a century, but ...